juriscraper/opinions/united_states/state/tex.py

```diff
@@ -14,298 +14,125 @@
             )
-            records_nr = self.find_element_by_xpath(
-                '//thead//*[contains(concat(" ", normalize-space(@class), " "), " rgInfoPart ")]/strong[1]'
+            self._set_parameters(view_state)
+            self.html = super()._download()
+
+        for row in self.html.xpath("//table[@class='rgMasterTable']/tbody/tr"):
+            # In texas we also have to ping the case page to get the name
+            # this is unfortunately part of the process.
+            self.url = row.xpath(".//a")[2].get("href")
+            self.seeds.append(self.url)
+            self.cases.append(
+                {
+                    "date": row.xpath(f".//td[2]")[0].text_content(),
+                    "docket": row.xpath(f".//td[5]")[0].text_content(),
+                    "url": row.xpath(".//a")[1].get("href"),
+                }
             )
-            html_pages = []
-            if records_nr:
-                self.records_nr = int(records_nr.text)
-            if nr_of_pages:
-                if nr_of_pages.text == "1":
-                    text = self.webdriver.page_source
-                    self.webdriver.quit()
-
-                    html_tree = html.fromstring(text)
-                    html_tree.make_links_absolute(self.url)
-
-                    remove_anchors = lambda url: url.split("#")[0]
-                    html_tree.rewrite_links(remove_anchors)
-                    html_pages.append(html_tree)
-                else:
-                    logger.info(
-                        "Paginating through %s pages of results."
-                        % nr_of_pages.text
-                    )
-                    logger.info("  Getting page 1")
-                    text = self.webdriver.page_source
-
-                    html_tree = html.fromstring(text)
-                    html_tree.make_links_absolute(self.url)
-
-                    remove_anchors = lambda url: url.split("#")[0]
-                    html_tree.rewrite_links(remove_anchors)
-                    html_pages.append(html_tree)
-
-                    for i in range(int(nr_of_pages.text) - 1):
-                        logger.info(f"  Getting page {i + 2}")
-                        next_page = self.find_element_by_class_name(
-                            "rgPageNext"
-                        )
-                        next_page.click()
-                        self.webdriver.implicitly_wait(5)
-
-                        text = self.webdriver.page_source
-
-                        html_tree = html.fromstring(text)
-                        html_tree.make_links_absolute(self.url)
-
-                        remove_anchors = lambda url: url.split("#")[0]
-                        html_tree.rewrite_links(remove_anchors)
-                        html_pages.append(html_tree)
-                    self.webdriver.quit()
-            return html_pages
 
     def _get_case_names(self):
-        def fetcher(url):
+        """Get case names using a deferring list."""
```

Comment:
This looks better, but I hesitate about your use of `self`:

 - You're using the class to pass seeds around. Could you get the seeds in a tidy way here instead of setting them up above? 
 - You're using self._download(), which means you have to use self.html, and self.url, and self.parameters. It's all a bit complex. Could you just not use `self` for any of that, and do a simple download instead? It'd be a lot easier and it wouldn't worry me about side effects of resetting self.html that we might be encountering someday.

Otherwise, looks like the right idea.

Reply:
I do use the self.html - the link is passed in for logging purposes

Reply:
I suppose the answer is always yes.

Reply:
as usual - it looks a bit nicer.  It also let me find - and cleanup - the naming convention for the case where the court just leaves off the State of Texas because it is presumed in the Court of Criminal Appeals.